src/model/store.coffee

```diff
@@ -59,6 +59,19 @@ class Request extends Varying
       # TODO: This should generate an object too. How does that happen?
       response
 
+  # Returns a promise for the deserialized request result. If the request
+  # fails, the promise is rejected with the request state.
+  asPromise: () ->
+    new Promise((resolve, reject) =>
+      this.reactNow((state) =>
+        if (state instanceof Request.state.type.Complete)
+          if (state instanceof Request.state.type.Success)
+            resolve(state.result)
+          else
+            reject(state)
```

Comment:
Requests may go through Progress state before reaching Success state, particularly on requests that hit 202-enabled endpoints.